catalogs/catalog-lakehouse-generic/src/main/java/org/apache/gravitino/catalog/lakehouse/lance/LanceTableOperations.java

```diff
@@ -100,45 +109,59 @@ public Table createTable(
     String location = properties.get(Table.PROPERTY_LOCATION);
     Preconditions.checkArgument(
         StringUtils.isNotBlank(location), "Table location must be specified");
-    Map<String, String> storageProps = LancePropertiesUtils.getLanceStorageOptions(properties);
+
+    // Extract creation mode from properties
+    CreationMode mode =
+        Optional.ofNullable(properties.get(LANCE_CREATION_MODE))
+            .map(
+                modeStr -> {
+                  try {
+                    return CreationMode.valueOf(modeStr);
+                  } catch (IllegalArgumentException e) {
+                    throw new IllegalArgumentException(
+                        "Invalid creation mode: "
+                            + modeStr
+                            + ". Supported modes are CREATE, EXIST_OK, "
+                            + "OVERWRITE.");
+                  }
+                })
+            .orElse(CreationMode.CREATE);
 
     boolean register =
-        Optional.ofNullable(properties.get(LanceTableDelegator.PROPERTY_LANCE_TABLE_REGISTER))
+        Optional.ofNullable(properties.get(LANCE_TABLE_REGISTER))
             .map(Boolean::parseBoolean)
             .orElse(false);
-    if (register) {
-      // If this is a registration operation, just create the table metadata without creating a new
-      // dataset
-      return super.createTable(
-          ident, columns, comment, properties, partitions, distribution, sortOrders, indexes);
+
+    // Handle EXIST_OK mode - check if table exists first
+    if (mode == CreationMode.EXIST_OK) {
```

Comment:
I remember that only modes `create` and `overwrite` are allowed for registering a table.

Reply:
I see. Let me update the code.